src/main/scala/org/constellation/gossip/GossipService.scala

```diff
@@ -0,0 +1,43 @@
+package org.constellation.gossip
+
+import cats.effect.Concurrent
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import org.constellation.gossip.sampling.PeerSampling
+import org.constellation.infrastructure.p2p.ClientInterpreter
+import org.constellation.schema.Id
+
+abstract class GossipService[F[_], A](selfId: Id, peerSampling: PeerSampling[F, Id])(implicit val F: Concurrent[F]) {
+
+  private val clients: Ref[F, Map[Id, ClientInterpreter[F]]] = Ref.unsafe(Map.empty[Id, ClientInterpreter[F]])
```

Comment:
Why do you need to store same ClientInterpreter for every Id? ClientInterpreter is a singleton afaik. 
IMO you should store here peer metadata (host, port, id) which will be applied to a proper api call (as providing a context into a run method)

Reply:
I mislooked ClientInterpreter with PeerClientMetadata :D